api/RPC.js

```diff
@@ -0,0 +1,174 @@
+'use strict';
+
+var Q = require('q');
+var _ = require('lodash');
+var scClient = require('socketcluster-client');
+var WAMPClient = require('wamp-socket-cluster/WAMPClient');
+var MasterWAMPServer = require('wamp-socket-cluster/MasterWAMPServer');
+var constants = require('../helpers/constants');
+
+var WsRPCServer = {
+
+	wsServer: null,
+	wampClient: new WAMPClient(),
+	scClient: scClient,
+	wsClientsConnectionsMap: {},
+
+	/**
+	 * @param {MasterWAMPServer} wsServer
+	 */
+	setServer: function (wsServer) {
+		this.wsServer = wsServer;
+	},
+
+	/**
+	 * @throws {Error} thrown if wsServer haven't been initialized before
+	 * @returns {MasterWAMPServer} wsServer
+	 */
+	getServer: function () {
+		if (!this.wsServer) {
+			throw new Error('WS server haven\'t been initialized!');
+		}
+		return this.wsServer;
+	}
+
+};
+
+/**
+ * @param {string} ip
+ * @param {number} port
+ * @returns {clientStub} {[string]: function} map where keys are all procedures registered
+ * @constructor
+ */
+function WsRPCClient (ip, port) {
+
+	if (!ip || !port) {
+		throw new Error('WsRPCClient needs ip and port to establish WS connection.');
+	}
+
+	var address = ip + ':' + port;
+	var socketDefer = WsRPCServer.wsClientsConnectionsMap[address];
+
+	//first time init || previously rejected
+	if (!socketDefer || socketDefer.promise.inspect().state === 'rejected') {
+		socketDefer = Q.defer();
+		WsRPCServer.wsClientsConnectionsMap[address] = socketDefer;
+	}
+
+	return this.clientStub(this.sendAfterSocketReadyCb(ip, port, socketDefer));
+}
+
+/**
+ * @param {object} options
+ * @param {Q.defer} socketDefer
+ */
+WsRPCClient.prototype.initializeNewConnection = function (options, socketDefer) {
+
+	var clientSocket = WsRPCServer.scClient.connect(options);
+
+	WsRPCServer.wampClient.upgradeToWAMP(clientSocket);
+
+	socketDefer.promise.initialized = true;
+
+	clientSocket.on('connect', function () {
+		if (!constants.externalAddress) {
```

Comment:
insted of `constants.externalAddress` maybe will be better `constants.ws.externalAddress`